File: src/collate.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jun 13 12:02:52 2023
Collates pre and post values in prep for sawtooth plots
@author: cognotrend
"""
import numpy as np

def collate(v1, v2):
    if v1.shape == v2.shape:
        collated = np.empty((v1.size * 2,), dtype=v1.dtype)
        collated[::2] = v1
        collated[1::2] = v2
    else:
        collated = -1
    return collated

File: src/matlab/collate.m
function [collated]=collate(v1,v2)
% collate.m collates two vectors of length m into a single vector of
% length 2m.
if length(v1)==length(v2)
    m=length(v1);
    collated = zeros(1,2*m);
    j=1;
    for i=1:m
        collated(j)=v1(i);
        collated(j+1)=v2(i);
        j=j+2;
    end
else
    collated=-1;
end
return
        